### PR TITLE
Allow empty sortWhitelist to not allow sorting at all.

### PR DESCRIPTION
**Pull request by dereuromark**
Resolves https://github.com/cakephp/cakephp/issues/6843

### CODE REVIEW

File: src/Controller/Component/PaginatorComponent.php [hunk 1]
```diff
@@ -319,7 +319,7 @@ public function validateSort(Table $object, array $options)
         }
 
         $inWhitelist = false;
-        if (!empty($options['sortWhitelist'])) {
+        if (isset($options['sortWhitelist']) && $options['sortWhitelist'] !== false) {
```
Comment (ADmad) [inline src/Controller/Component/PaginatorComponent.php]: Wouldn't just isset check be enough?  Given that the default behavior is no whitelist I don't see why anyone would bother setting it to `false` to achieve the same.

Comment (dereuromark) [inline src/Controller/Component/PaginatorComponent.php]: Probably. I can modify accordingly.

**Comment by ADmad** [comment]
Could you please also update the docblock above paginate() stating this new feature.

**Comment by dereuromark** [comment]
I wonder if we should backport this to 2.7.
Currently there it is `If empty array all keys are allowed`.

**Comment by lorenzo** [comment]
This looks good to me :+1:

**Comment by markstory** [comment]
Nice work @dereuromark

**Pull request merged**